src/components/app/App.jsx

```diff
@@ -125,17 +173,54 @@ export const App = () => {
 
     //
 
+    useEffect( () => {
+
+        const toggleOpen = () => {
+
+            if ( state.settings.opened ) return;
+            ioManager.dispatchEvent( new CustomEvent( 'CloseAllPanels' ) );
+
+            const newValue = ! state.world.opened;
+
+            if ( newValue ) {
+
+                cameraManager.exitPointerLock();
+
+            } else {
+
+                cameraManager.requestPointerLock();
+
+            }
+
+            setState({ ...state, world: { ...state.world, opened: newValue } });
+
+        };
+
+        ioManager.addEventListener( 'ToggleWorldPanel', toggleOpen );
```

Comment:
IO Manager should not know about panels. This state belongs in `game.js`, or a UI manager.